bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/WDChannels.java

```diff
@@ -51,43 +56,101 @@ public synchronized WorkerDataChannel getWorkerDataChannel(String name) {
         return channel;
     }
 
-    public Object tryTakeData(Strand strand, List<String> channels) throws Throwable {
+    public Object receiveDataMultipleChannels(Strand strand, ReceiveField[] receiveFields, Type targetType)
+            throws Throwable {
```

Comment:
Do we need to throw Throwable here?

Reply:
Then can we changed to BError?

Reply:
We catch the panic as `Throwable` during the scheduler execution and keep it in the `WorkerDataChannel`. 
Then, we throw the panic when executing the receive action. 
https://github.com/ballerina-platform/ballerina-lang/blob/bb0abfc93b4a897d3318c702c9793b3011b69c8d/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java#L321

If we change to catch only `BError` in the `Scheduler`, we might miss the Runtime exceptions and crashes.